byway/src/main/webapp/routepage.html

```diff
@@ -0,0 +1,49 @@
+<!DOCTYPE html>
+<html>
+<head>
+  <meta charset="UTF-8">
+  <title>Route Generator</title>
+  <link rel="stylesheet" href="/bootstrap/bootstrap.min.css" crossorigin="anonymous">
+  <script src="https://polyfill.io/v3/polyfill.min.js?features=default"></script>
+  <script
+      src="https://maps.googleapis.com/maps/api/js?key=AIzaSyBg6D1U0U2L7VlbKH7Bkh-ds6wdUaLzroM&callback=initMap&libraries=&v=weekly"
+      defer></script>
+  <script src="map.js"></script>
+  <script src="routepage.js"></script>
+  <link rel="stylesheet" href="style.css">
+</head>
+<body onLoad="getRecs()">
```

Comment:
Note that body onLoad waits for all page content to be fully loaded before executing (including loading images, fonts etc. on the page). This tends to make the page seem unresponsive initially, especially for pages with many images, since setup JS functions may not be run for a couple seconds as all the fonts and such get pulled in.

There are a couple ways to avoid this: one is to put this action in a `<script>getRecs()</script>` tag at the bottom of the body (https://stackoverflow.com/a/32998549). Another, possibly better option is to just listen for the `DOMContentLoaded` event with `document.addEventListener`, as seen here: https://developer.mozilla.org/en-US/docs/Web/API/Document/DOMContentLoaded_event

The `DOMContentLoaded` event has some advantages: you can configure it directly from your included script, so you don't have to put a related tag in the HTML, and since it's using the `addEventListener` mechanism instead of `onLoad`, you can actually have multiple callback listening for the same event. This is good if you have multiple independent setup things that should run for your page. For example if you have a common script that sets up headers and footers, and a page-specific script that sets up some element of this particular page, they can both add listeners for the same event, and you don't have to worry about having one of them call the other.

Reply:
Okay, I changed it to the second suggested approach. Thanks for the links I'll probably need to read up more on this as we use multiple js files.